pipelines/batch/src/main/java/com/google/fhir/analytics/ParquetMerger.java

```diff
@@ -243,7 +242,7 @@ private static GenericRecord findLastRecord(
     if (numRec > 1) {
       numDuplicates.inc();
     }
-    if (numRec > 2) {
+    if (numRec > 2 && lastRecord != null && lastRecord.get(ID_KEY) != null) {
```

Comment:
nit: please add a comment saying that it is guaranteed for `lastRecord` not to be `null` if `numRec > 1`. Also why do we need the `lastRecord.get(ID_KEY) != null` check? Wouldn't it simply add `null` as a text to the `warn` message below?